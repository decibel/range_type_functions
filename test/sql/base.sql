\set ECHO none

\i test/pgxntool/setup.sql
SET search_path = :install_schema, tap;
SELECT plan(
    0
    + 4 -- to_range
    + 3 -- element_range_comp
    + 2 -- is_singleton
);

create type textrange as range (subtype = text);
create type textrange_c as range (subtype = text, collation = "C");

set datestyle = 'ISO';

select is(
    to_range(null::int4range, 4,5,'[]')
    , '[4,5]'::int4range
);

select is(
    to_range(null::daterange, '2015-01-01'::date,'2016-01-1')
    , '[2015-1-1,2016-1-1)'::daterange
);

select is(
    to_range(null::int4range, 4)
    , '[4,4]'::int4range
);

select is(
    to_range(null::daterange, '2015-01-01'::date)
    , '[2015-1-1,2015-1-1]'::daterange
);

select is(
    element_range_comp('[10,100]'::int4range, 4)
    , -1::smallint
    , $$element_range_comp('[10,100]'::int4range, 4)$$
);

select is(
    element_range_comp('[10,100]'::int4range, 10)
    , 0::smallint
    , $$element_range_comp('[10,100]'::int4range, 10)$$
);

select is(
    element_range_comp('[10,100]'::int4range, 110)
    , 1::smallint
    , $$element_range_comp('[10,100]'::int4range, 110)$$
);

select is(
    is_singleton('[4,5)'::int4range)
    , true
    , $$is_singleton('[4,5)'::int4range)$$
);

select is(
    is_singleton('[4,5]'::int4range)
    , false
    , $$is_singleton('[4,5]'::int4range)$$
);

\set ECHO ALL

select range_merge('[4,5]'::int4range,'[9,10]'::int4range);

select get_lower_bound_condition_expr('empty'::int4range);
select get_lower_bound_condition_expr('(,)'::int4range);
select get_lower_bound_condition_expr('(4,5]'::int4range);
select get_lower_bound_condition_expr('(4,5]'::int4range,'y.z');
select get_lower_bound_condition_expr('[4,5]'::int4range,'y.z');
select get_lower_bound_condition_expr('[4,5]'::int4range,format('%I.%I','my schema','my ColuMnaME'));
select get_upper_bound_condition_expr('empty'::int4range);
select get_upper_bound_condition_expr('(,)'::int4range);
select get_upper_bound_condition_expr('[4,5)'::int4range,'y.z');
select get_upper_bound_condition_expr('[4,5]'::int4range,'y.z');
select get_upper_bound_condition_expr('[ABEL,BAKER)'::textrange,'y.z');
select get_upper_bound_condition_expr('[ABEL,BAKER)'::textrange_c,'y.z');
select get_bounds_condition_expr('empty'::int4range);
select get_bounds_condition_expr('(,)'::int4range);
select get_bounds_condition_expr('(4,5]'::int4range);
select get_bounds_condition_expr('(4,5]'::int4range,'y.z');
select get_bounds_condition_expr('[4,5]'::int4range,'y.z');
select get_bounds_condition_expr('[4,5)'::int4range,'y.z');
select get_bounds_condition_expr('[4,5]'::int4range,'y.z');

select get_collation_expr(null::int4range);
select get_collation_expr(null::textrange);
select get_collation_expr(null::textrange_c);


select get_subtype_element_expr('(4,5]'::int4range);
select get_subtype_element_expr('[ABEL,BAKER)'::textrange_c);

select get_bound_expr(null::int4range,'4');
select get_bound_expr(null::daterange,'1991-09-23');
select get_bound_expr(null::textrange,'ABEL');
select get_bound_expr(null::textrange_c,'ABEL');

\i test/pgxntool/finish.sql
