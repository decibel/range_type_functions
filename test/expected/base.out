\set ECHO none
1..12
ok 1
ok 2
ok 3
ok 4
ok 5 - element_range_comp('[10,100]'::int4range, 4)
ok 6 - element_range_comp('[10,100]'::int4range, 10)
ok 7 - element_range_comp('[10,100]'::int4range, 110)
ok 8 - is_singleton('[4,5)'::int4range)
ok 9 - is_singleton('[4,5]'::int4range)
ok 10 - get_collation_expr(null::int4range)
ok 11 - get_collation_expr(null::textrange)
ok 12 - get_collation_expr(null::textrange_c)
select range_merge('[4,5]'::int4range,'[9,10]'::int4range);
[4,11)
select get_lower_bound_condition_expr('empty'::int4range);
false
select get_lower_bound_condition_expr('(,)'::int4range);
true
select get_lower_bound_condition_expr('(4,5]'::int4range);
x >= '5'::integer
select get_lower_bound_condition_expr('(4,5]'::int4range,'y.z');
y.z >= '5'::integer
select get_lower_bound_condition_expr('[4,5]'::int4range,'y.z');
y.z >= '4'::integer
select get_lower_bound_condition_expr('[4,5]'::int4range,format('%I.%I','my schema','my ColuMnaME'));
"my schema"."my ColuMnaME" >= '4'::integer
select get_upper_bound_condition_expr('empty'::int4range);
false
select get_upper_bound_condition_expr('(,)'::int4range);
true
select get_upper_bound_condition_expr('[4,5)'::int4range,'y.z');
y.z < '5'::integer
select get_upper_bound_condition_expr('[4,5]'::int4range,'y.z');
y.z < '6'::integer
select get_upper_bound_condition_expr('[ABEL,BAKER)'::textrange,'y.z');
y.z COLLATE "default" < 'BAKER'::text COLLATE "default"
select get_upper_bound_condition_expr('[ABEL,BAKER)'::textrange_c,'y.z');
y.z COLLATE "C" < 'BAKER'::text COLLATE "C"
select get_bounds_condition_expr('empty'::int4range);
false
select get_bounds_condition_expr('(,)'::int4range);
true
select get_bounds_condition_expr('(4,5]'::int4range);
x >= '5'::integer and x < '6'::integer
select get_bounds_condition_expr('(4,5]'::int4range,'y.z');
y.z >= '5'::integer and y.z < '6'::integer
select get_bounds_condition_expr('[4,5]'::int4range,'y.z');
y.z >= '4'::integer and y.z < '6'::integer
select get_bounds_condition_expr('[4,5)'::int4range,'y.z');
y.z >= '4'::integer and y.z < '5'::integer
select get_bounds_condition_expr('[4,5]'::int4range,'y.z');
y.z >= '4'::integer and y.z < '6'::integer
select get_subtype_element_expr('(4,5]'::int4range);
x
select get_subtype_element_expr('[ABEL,BAKER)'::textrange_c);
x COLLATE "C"
select get_bound_expr(null::int4range,'4');
'4'::integer
select get_bound_expr(null::daterange,'1991-09-23');
'1991-09-23'::date
select get_bound_expr(null::textrange,'ABEL');
'ABEL'::text COLLATE "default"
select get_bound_expr(null::textrange_c,'ABEL');
'ABEL'::text COLLATE "C"
\i test/pgxntool/finish.sql
SELECT finish();
\echo # TRANSACTION INTENTIONALLY LEFT OPEN!
# TRANSACTION INTENTIONALLY LEFT OPEN!
